src/main/PersistentState.cpp

```diff
@@ -37,7 +37,7 @@ PersistentState::PersistentState(Application& app) : mApp(app)
 void
 PersistentState::deleteTxSets(std::unordered_set<Hash> hashesToDelete)
 {
-    // No need for soci::transaction here; try to delete as much as we can
+    soci::transaction tx(mApp.getDatabase().getSession());
```

Comment:
curious, did you observe perf degradation in a profiler?

Reply:
I didn't look for this particular one as this tends to be quite unpredictable. without this you have one sql transaction per txset. even on sqlite this means one fsync for each. So this new version is at least as fast as the original one and we don't have to maintain a comment.